File: Maths/Geometry/Polar_Cartesian_Conversion/Polar_Cartesian_Conversion.go
/*
	This code takes input r and θ in Polar Coordinates and converts them into cartesian coordinates.
	Language : Golang
	Author : Pulkit Sapra
*/

package main

//Package fmt implements formatted I/O with functions analogous to C's printf and scanf. 
import (
	"fmt"
	"math"
	)

func convert_to_cartesian(r float64, theta float64){
	var x,y float64

	//convert degree to radians
	theta = theta*math.Pi/180 

	//Conversion of polar coord. to Cartesian
	x = r * math.Cos(theta)
	y = r * math.Sin(theta) 

	fmt.Printf("x = %0.3f, y = %0.3f\n", x , y)

}

// Main Function
func main() {
	fmt.Print("Please enter radius(r) and theta(in degree) : ")
	
	// Declaring variables and their types.
	var r,theta float64

	// Taking input for r adn theta
	fmt.Scan(&r)
	fmt.Scan(&theta)

	//Function call
	convert_to_cartesian(r,theta)
	
}

File: Maths/Geometry/Polar_Cartesian_Conversion/Polar_Cartesian_Conversion.java
import java.util.Scanner;

public class PolarCartesian {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Input radius: ");
        double radius = sc.nextDouble();
        System.out.print("Input theta in degrees: ");
        double theta = sc.nextDouble();

        // converting theta from degrees to radians
        theta = theta * Math.PI / 180f;

        double x = radius * Math.cos(theta);
        double y = radius * Math.sin(theta);

        System.out.printf("Cartesian coordinates: (%5.3f, %5.3f)", x, y);
    }
}
